### PR TITLE
feat(shell): add shell package with app-grid

### LINKED ISSUE #462: Add shell package
Add shell package with app-grid to serve app scenarios

### PR DESCRIPTION
**Pull request by nicholasrice**
closes #462

### CODE REVIEW

File: packages/fast-shell-react-msft/src/pane/pane.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    public componentWillUnmount(): void {
+        window.removeEventListener("resize", this.onWindowResize);
+    }
+
+    /**
+     * Handle when component updates
+     */
+    public componentDidUpdate(prevProps: PaneProps, prevState: IPaneState): void {
+        if ( this.state.resizing && !prevState.resizing ) {
+            document.addEventListener("mouseup", this.onMouseUp);
```
Comment (chrisdholt) [inline packages/fast-shell-react-msft/src/pane/pane.tsx]: Do we need to check for the DOM on these prior to accessing the document? Assume we would need a check when adding event listeners above as well?

suggest https://www.npmjs.com/package/exenv-es6 :)

File: packages/fast-shell-react-msft/src/utilities/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Converts a number to a string with the number represented as a pixel value
+ */
+export function toPx(value: number): string {
+    return `${value}px`;
+}
```
Comment (chrisdholt) [inline packages/fast-shell-react-msft/src/utilities/index.ts]: Doesn't this already exist in the fast-jss-utilities package you are importing?

File: packages/fast-layouts-react/package.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{
+  "name": "@microsoft/fast-layouts-react",
+  "version": "1.0.0",
+  "description": "FAST Microsoft app shell written in React",
```
Comment (chrisdholt) [inline packages/fast-layouts-react/package.json]: @nicholasrice probably want to broaden the description to just layouts with the change

File: packages/fast-development-site-react/src/components/site/action-bar.tsx [hunk 4]
```diff
@@ -83,6 +83,8 @@ function menuButtonBase(): ICSSRules<IDevSiteDesignSystem> {
 const styles: ComponentStyles<IActionBarClassNameContract, IDevSiteDesignSystem> = {
     actionBar: {
         display: "flex",
+        flexBasis: "100%",
+        borderBottom: `${toPx(1)} solid #CCCCCC`
```
Comment (nalogiudice) [inline packages/fast-development-site-react/src/components/site/action-bar.tsx]: Not a design system color, or cannot be made using Chroma?

Comment (nicholasrice) [reply packages/fast-development-site-react/src/components/site/action-bar.tsx]: @nalogiudice I'm not following - what are you asking for?

File: packages/fast-layouts-react/src/pane/pane.tsx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        flex: "0 1 auto",
+        display: "flex",
+        flexDirection: "column"
+    },
+    pane_resizeHandle: {
+        position: "absolute",
+        padding: "0",
+        opacity: "0",
+        top: "0",
+        width: "8px",
```
Comment (nalogiudice) [inline packages/fast-layouts-react/src/pane/pane.tsx]: toPx() ?

File: packages/fast-development-site-react/src/components/site/action-bar.tsx (hunk 4 above)
Comment (nalogiudice) [reply packages/fast-development-site-react/src/components/site/action-bar.tsx]: Take a look at checkbox - it uses Chroma.mix to achieve grays using the design system colors. I'm wondering if we should be doing that everywhere

**Pull request merged**